packages/shared/src/components/filters/MyFeedHeading.tsx

```diff
@@ -103,46 +109,64 @@ function MyFeedHeading({
   });
   const isRefreshing = feedQueryState?.status === 'success' && isFetchingFeed;
 
-  return (
-    <>
-      {forceRefresh && (
-        <Button
-          size={
-            shouldUseMobileFeedLayout ? ButtonSize.Small : ButtonSize.Medium
-          }
-          variant={ButtonVariant.Float}
-          className="mr-auto"
-          onClick={onRefresh}
-          icon={<RefreshIcon />}
-          iconPosition={
-            shouldUseMobileFeedLayout ? ButtonIconPosition.Right : undefined
-          }
-          loading={isRefreshing}
-        >
-          {!isMobile ? 'Refresh feed' : null}
-        </Button>
-      )}
-      <AlertPointer {...alertProps}>
-        <Button
-          size={
-            shouldUseMobileFeedLayout ? ButtonSize.Small : ButtonSize.Medium
-          }
-          variant={ButtonVariant.Float}
-          className={classNames(
-            'mr-auto',
-            shouldHighlightFeedSettings && 'highlight-pulse',
-          )}
-          onClick={onClick}
-          icon={<FilterIcon />}
-          iconPosition={
-            shouldUseMobileFeedLayout ? ButtonIconPosition.Right : undefined
-          }
-        >
-          {!isMobile ? 'Feed settings' : null}
-        </Button>
-      </AlertPointer>
-    </>
-  );
+  const RenderFeedActions = () => {
+    return (
+      <>
+        {forceRefresh && (
+          <Button
+            size={
+              shouldUseMobileFeedLayout ? ButtonSize.Small : ButtonSize.Medium
+            }
+            variant={ButtonVariant.Float}
+            className="mr-auto"
+            onClick={onRefresh}
+            icon={<RefreshIcon />}
+            iconPosition={
+              shouldUseMobileFeedLayout ? ButtonIconPosition.Right : undefined
+            }
+            loading={isRefreshing}
+          >
+            {!isMobile && !useNewMobileLayout ? 'Refresh feed' : null}
+          </Button>
+        )}
+        <AlertPointer {...alertProps}>
+          <Button
+            size={
+              shouldUseMobileFeedLayout ? ButtonSize.Small : ButtonSize.Medium
+            }
+            variant={ButtonVariant.Float}
+            className={classNames(
+              'mr-auto',
+              shouldHighlightFeedSettings && 'highlight-pulse',
+            )}
+            onClick={onClick}
+            icon={<FilterIcon />}
+            iconPosition={
+              shouldUseMobileFeedLayout ? ButtonIconPosition.Right : undefined
+            }
+          >
+            {!isMobile && !useNewMobileLayout ? 'Feed settings' : null}
+          </Button>
+        </AlertPointer>
+      </>
+    );
+  };
+
+  if (useNewMobileLayout) {
```

Comment:
We can use conditionalWrapper component for this with no need to extract RenderFeedActions into a component as done now